### PR TITLE
chore(frontend): touch up components and pages for improved UI and code consistency

### PR DESCRIPTION
**Pull request by tmcrockett252**
- Refined styles and layout for better user experience
- Cleaned up code and improved readability in multiple components/pages
- Fixed minor UI inconsistencies and alignment issues
- Updated comments and structure for maintainability
- Waiting for authentication / Clerk / API to be merged to fully update the buttons

### CODE REVIEW

**Comment by tmcrockett252** [comment]
Changes:

- Split main event form into EventFormFields, GuestInput, and GuestList components to improve code maintainability, readability, and reusability.
- Centralized API integration and state management in EventForm
- Enhanced code readability and reusability for future backend and authentication integration

File: event-planner/frontend/src/components/eventForm/EventForm.tsx [hunk 1]
```diff
@@ -9,60 +12,155 @@ const EventForm: React.FC = () => {
...
         const eventData = {
             title,
             description,
             location,
-            owner,
+            owner, // Will be replaced with Clerk user ID/name when integrated
             time,
-            imageUrl
+            imageUrl,
+            guests,
```
Comment (Boyce007) [inline event-planner/frontend/src/components/eventForm/EventForm.tsx]: We want our guests to onlly have an input option when an event is private. There should be some kind os conditional rendering so public events have no guest list but private events you can invite to

File: event-planner/frontend/src/components/eventForm/EventFormFields.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  owner,
+  setOwner,
+  time,
+  setTime,
+  imageUrl,
+  setImageUrl,
+  isSubmitting
+}) => (
+  <>
+    <div className="form-group">
```
[the remark this answers is not in the document]
Comment (tmcrockett252) [reply event-planner/frontend/src/components/eventForm/EventFormFields.tsx]: I will attempt to do this on another branch so this one isn't jumbled up.

**Review by Boyce007 — DISMISSED** [review]

**Review by JoyBrown252 — COMMENTED** [review]
**1. Fix Code Duplication in EventGuestController.java**
The helper methods buildUserDTO() and buildGuestDTOs() duplicate existing functionality:

// REMOVE these duplicate methods (lines 56-77):
private com.api.demo.dtos.UserDTO buildUserDTO(com.api.demo.models.User user) {
    if (user == null) return null;
    return com.api.demo.dtos.UserDTO.builder()
        .id(user.getId())
        .name(user.getName())
        .email(user.getEmail())
        .build();
}

private java.util.Set<com.api.demo.dtos.UserDTO> buildGuestDTOs(
    java.util.Set<com.api.demo.models.EventGuest> eventGuests) {
    // ... duplicate mapping logic
}

// REPLACE with:
import com.api.demo.dtos.DTOConverter; // Assuming this exists from PR #85

// In createEventWithGuests method:
public ResponseEntity<EventDTO> createEventWithGuests(
    @PathVariable Long organizerId, @RequestBody CreateEventWithGuestsRequest request) {
    EventModel createdEvent = eventGuestService.createEventWithGuests(
        organizerId, request.getEvent(), request.getGuestEmails());
    
    EventDTO eventDTO = DTOConverter.mapToDTO(createdEvent); // Use existing converter
    return ResponseEntity.status(HttpStatus.CREATED).body(eventDTO);
}

**2. Add Missing imageURL Field Validation**
Add validation for the new imageURL field in EventModel:

// In EventModel.java, add validation:
@Column(name = "image_url")
@URL(message = "Image URL must be valid")
private String imageURL;


**3. Fix Inconsistent CSS Architecture**
Remove font-family declarations from individual CSS files since it's now in App.css:

/* REMOVE from Event.css (line 53): */
font-family: 'Satoshi', sans-serif;

/* REMOVE from EventList.css (lines 5 and 15): */
font-family: 'Satoshi', sans-serif;

/* REMOVE from any other component CSS files with this declaration */


**4. Add Missing Test Coverage**
The test updates in EventGuestControllerTest.java need additional edge cases:

@Test
void shouldTestCreateEventWithGuests_NullImageURL() {
    // Test when imageURL is null
    mockEvent.setImageURL(null);
    // ... rest of test
    assertNull(response.getBody().getImageURL());
}

@Test
void shouldTestCreateEventWithGuests_NoGuests() {
    // Test when eventGuests is empty
    mockEvent.setEventGuests(new HashSet<>());
    // ... rest of test
    assertTrue(response.getBody().getGuests().isEmpty());
}


**5. Secure Environment Variables**

# CHECK: Verify no .env files are already committed
git log --all --full-history -- "**/.env"

# IF FOUND: Remove from history using git-filter-branch or BFG


**6. Fix Missing Fields in DTO Mapping
Several controllers are missing the imageURL field mapping - ensure consistency:**

// In UserController.java updateUserEvent method:
// ADD missing .id() field (line 133 is added but verify it's complete):
EventDTO eventDTO = EventDTO.builder()
    .title(updatedEvent.getTitle())
    .description(updatedEvent.getDescription())
    .startTime(updatedEvent.getStartTime())
    .eventType(updatedEvent.getIsPublic() ? "Community" : "Private")
    .organizer(organizerDTO)
    .guests(guests)
    .imageURL(updatedEvent.getImageURL())
    .id(updatedEvent.getId()) // This was added but verify placement
    .build();

**Comment by JoyBrown252** [comment]
1. Remove Code Duplication in EventGuestController.java:

// REMOVE these duplicate methods:
private com.api.demo.dtos.UserDTO buildUserDTO(com.api.demo.models.User user) {
    if (user == null) return null;
    return com.api.demo.dtos.UserDTO.builder()
        .id(user.getId())
        .name(user.getName())
        .email(user.getEmail())
        .build();
}

private java.util.Set<com.api.demo.dtos.UserDTO> buildGuestDTOs(
    java.util.Set<com.api.demo.models.EventGuest> eventGuests) {
    // ... duplicate mapping logic
}

// REPLACE with proper DTO converter usage

2. Add Missing imageURL Field Validation
The new imageURL field in EventModel.java needs proper validation

// In EventModel.java, add validation:
@Column(name = "image_url")
@URL(message = "Image URL must be valid")
private String imageURL;

3. Fix CSS Architecture Inconsistency
Remove redundant font-family declarations from individual CSS files since it's now centralized in App.css:

/* REMOVE from Event.css (line 53): */
font-family: 'Satoshi', sans-serif;

/* REMOVE from EventList.css (lines that had font-family): */
font-family: 'Satoshi', sans-serif;

4. Add Missing Test Coverage
The EventGuestControllerTest.java needs additional edge cases:

@Test
void shouldTestCreateEventWithGuests_NullImageURL() {
    // Test when imageURL is null
    mockEvent.setImageURL(null);
    // ... rest of test
    assertNull(response.getBody().getImageURL());
}

@Test 
void shouldTestCreateEventWithGuests_EmptyGuests() {
    // Test when eventGuests is empty
    mockEvent.setEventGuests(new HashSet<>());
    // ... rest of test
    assertTrue(response.getBody().getGuests().isEmpty());
}

5. Complete DTO Field Mapping
Ensure all controllers consistently map the imageURL field. The UserController.java has the field added but verify complete mapping in all DTO builders.


6. Verify Environment Variable Security
Check that no .env files were accidentally committed in git history:

# Run this command to check:
git log --all --full-history -- "**/.env"

# If found, remove from history using git-filter-branch or BFG
